apps/services/notifications/src/app/modules/notifications/consumer.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Inject, Injectable, OnModuleDestroy } from '@nestjs/common'
+import {
+  Message as SqsMessage,
+  ReceiveMessageCommand,
+  DeleteMessageBatchCommand,
+} from '@aws-sdk/client-sqs'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { CONNECTION_PROVIDER } from '../../../constants'
+import { SqsChannel } from '../../../types'
+import { MessageHandlerService } from './messageHandler.service'
+
+@Injectable()
+export class NotificationConsumerService implements OnModuleDestroy {
+  private currentBatchProcessing: Promise<void> = Promise.resolve()
+  private shutdown = false
+
+  constructor(
+    private readonly messageHandler: MessageHandlerService,
+    @Inject(CONNECTION_PROVIDER)
+    private readonly queue: SqsChannel,
+    @Inject(LOGGER_PROVIDER)
+    private readonly logger: Logger,
+  ) {}
+
+  async onModuleDestroy() {
+    this.shutdown = true
+
+    this.logger.info('Waiting for workers to finish')
+    await this.currentBatchProcessing
+
+    this.logger.info('Closing connection')
+    this.queue.client.destroy()
+  }
+
+  async run(): Promise<void> {
+    this.logger.info('Starting consumer')
```

Comment:
Maybe it's better if these messages better reflect each other, one might not instantly equate `Closing connection` with `Shutting down consumer`

Reply:
I think this describes pretty well what is actually being done, but I'm open to suggestions :)

Reply:
I think the messages describe what is being done just fine, I'm thinking when someone who does not know this app is debugging the logs they see `Starting consumer` but the shutting down message is `Closing connection`.
My concern is that it might not be obvious that the application is done running, therefor it might be more clear to have a message that better reflects the start message e.g. `Shutting down consumer`